docs/how_to_auxiliary.rst

```diff
@@ -0,0 +1,243 @@
+and **deleted** at test teardown by calling :py:meth:`~pykiso.interfaces.thread_auxiliary.AuxiliaryInterface.delete_instance`.
+
+These methods set the :py:attr:`is_instance attribute <pykiso.auxiliary.AuxiliaryCommon.is_instance>` that
+indicated if the auxiliary is running correctly.
+
+Concurrent auxiliary execution
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The execution of concurrent auxiliaries (i.e. inheriting from :py:class:`~pykiso.interfaces.thread_auxiliary.AuxiliaryInterface`
+or :py:class:`~pykiso.interfaces.mp_auxiliary.MpAuxiliaryInterface`) is
+handled by the interfaces' :py:meth:`run method <pykiso.interfaces.thread_auxiliary.AuxiliaryInterface.run>`.
+
+Each command execution is handled in a thread-safe way by getting values from an input queue and
+returning the command result in an output queue.
+
+Auxiliary run
+^^^^^^^^^^^^^
+
+Each time the execution is entered, the following actions are performed:
+
+1. Verify if a request is available in the input queue
+
+  #. If the command message is "create_auxiliary_instance" and the auxiliary is not created yet,
+     call the ``_create_auxiliary_instance`` method and put a boolean corresponding to the success
+     of the command processing in the output queue. This command message is put in the queue
+     at test setup.
+
+  #. If the command message is "delete_auxiliary_instance" and the auxiliary is created, call
+     the ``_delete_auxiliary_instance`` method and put a boolean corresponding the success of the
+     command processing in the output queue. This command message is put in the queue
+     at test teardown.
+
+  #. If the command message is a tuple of 3 elements starting with "command", then a custom command has to
+     be executed. This custom command has to be implemented in the ``_run_command`` method.
+
+  #. If the command message is "abort" and the auxiliary is created, call the ``_abort_command``
+     method and put a boolean corresponding the success of the command processing in the output queue.
+
+2. Verify if a Message is available for reception
+
+  #. Call the auxiliarie's ``_receive_message`` method
+  #. If something is returned, put it in the output queue, otherwise repeat this execution cycle.
+
+Implement an Auxiliary
+~~~~~~~~~~~~~~~~~~~~~~
+
+Common auxiliary methods
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+All of the above described Auxiliary interfaces require the same abstract methods
+to be implemented:
+
+- ``_create_auxiliary_instance``: handle the auxiliary creation. Minimal actions to perform are
+  opening the attached ``CChannel``, to which can be added actions such as flashing the device under test,
+  perform security related operations to allow the communication, etc.
+- ``_delete_auxiliary_instance``: handle the auxiliary deletion. This method is the counterpart of
+  ``_create_auxiliary_instance`` and should at least close the opened ``CChannel``.
+

```

Comment:
I think something like "_delete_auxiliary_instance" needs to be done in a way that "_create_auxiliariy_instance" can be called again without side effects.
is missing